src/Castle.Windsor/Core/Internal/Locking/Lock.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2004-2009 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Core.Internal.Locking
```

Comment:
Since you can't put these in `Castle.Core.Internal` since they would conflict with the ones still in the `Castle.Core` assembly, I think it would be better to put them into `Castle.MicroKernel.Internal` which already exists.